chatkit-core/src/test/kotlin/com/pusher/chatkit/rooms/state/JoinedRoomTest.kt

```diff
@@ -0,0 +1,32 @@
+package com.pusher.chatkit.rooms.state
+
+import assertk.assertThat
+import assertk.assertions.isNotNull
+import com.pusher.chatkit.state.ChatkitState
+import com.pusher.chatkit.state.JoinedRoom
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class JoinedRoomTest : Spek({
+
+    describe("given no rooms") {
+        val givenState = ChatkitState(joinedRoomsState = JoinedRoomsState(emptyMap(), emptyMap()))
+
+        describe("when one new room with unread counts is received") {
```

Comment:
```suggestion
        describe("when joined room") {
```

Reply:
hmm `when joined room` doesn't make grammatical sense - how about `when a joined room is received`?

Reply:
the received part from the first test around reducer corresponds to the event/action name: `JoinedRoomsReceived` so it would be good to avoid putting received everywhere as we have more specific verbs for other events then the initial state one. how about "when a room is joined"?

Reply:
That makes more grammatical sense. Have updated this to read `when a room is joined` 👍